app/knight/knight_class.py

```diff
@@ -0,0 +1,51 @@
+class Knight:
+    protection = 0
+    list_of_knights = {}
+
+    def __init__(self, knight: object) -> None:
+        # Major values
+        self.name = knight["name"]
+        self.power = knight["power"]
+        self.hp = knight["hp"]
+
+        # Adding properties to addition features
+        self.search_addition_armor(knight["armour"])
+        self.add_power_of_weapon(knight["weapon"])
+        self.add_properties_of_potion(knight["potion"])
+
+        # Adding all knights in the class list
+        self.list_of_knights[self.name] = self
```

Comment:
When referring to class-level attributes from `__init__`, use `Knight.list_of_knights[self.name] = ...` instead of `self.list_of_knights[self.name] = ...` because `list_of_knights` does not belong to a specific instance, it belongs to the class itself.

Reply:
+